fmc_objects.tf

```diff
@@ -983,28 +979,24 @@ resource "fmc_ipv6_address_pools" "module" {
 ######
 
 locals {
-  map_network_objects = merge({
-    for item in flatten([
-      for domain_key, domain_value in local.resource_hosts :
-      flatten([for item_key, item_value in domain_value.items : {
-        name        = item_key
-        id          = fmc_hosts.module[domain_key].items[item_key].id
-        type        = fmc_hosts.module[domain_key].items[item_key].type
-        domain_name = domain_key
-      }])
-    ]) : item.name => item if contains(keys(item), "name")
-    },
-    {
-      for item in flatten([
-        for domain_key, domain_value in local.data_hosts :
-        flatten([for item in keys(domain_value.items) : {
-          name        = item
-          id          = data.fmc_hosts.module[domain_key].items[item].id
-          type        = data.fmc_hosts.module[domain_key].items[item].type
-          domain_name = domain_key
-        }])
-      ]) : item.name => item if contains(keys(item), "name")
-    },
+  map_network_objects = merge(
+
+    # Hosts - bulk mode outputs
+    local.hosts_bulk ? merge([
+      for domain, hosts in fmc_hosts.module : {
+        for host_name, host_values in hosts.items : host_name => { id = host_values.id, type = host_values.type }
+      }
+    ]...) : {},
```

Comment:
Wouldn't this create conflicts if we would have two hosts with the same name in two different domains? Should we standardize on a "<domain>/<host>" key?